standard/game.lua

```diff
@@ -61,78 +65,135 @@ function Game.isValid(game)
 end
 
 ---Fetches the raw data for a given game
----@param args? {game: string?, useDefault: boolean?}
+---@param options? {game: string?, useDefault: boolean?}
 ---@return table
-function Game.raw(args)
-	local identifier = Game.toIdentifier(args)
+function Game.raw(options)
+	local identifier = Game.toIdentifier(options)
 	if not identifier then
 		return {}
 	end
 
 	return GamesData[identifier] or {}
 end
 
+---Fetches all valid game identifiers, potentially ordered
+---@param options? {ordered: boolean?}
+---@return table
+function Game.listGames(options)
+	options = options or {}
+
+	local gamesList = Array.extractKeys(GamesData or {})
+	if Logic.readBool(options.ordered) and Array.all(gamesList, function(gameIdentifier)
+				return tonumber(GamesData[gameIdentifier].order)
+			end) then
+		return Array.sortBy(gamesList, function(gameIdentifier)
+				return tonumber(GamesData[gameIdentifier].order)
+			end)
+	end
+
+	return gamesList
+end
+
 ---Fetches the abbreviation for a given game
----@param args? {game: string?, useDefault: boolean?}
+---@param options? {game: string?, useDefault: boolean?}
 ---@return string?
-function Game.abbreviation(args)
-	return Game.raw(args).abbreviation
+function Game.abbreviation(options)
+	return Game.raw(options).abbreviation
 end
 
 ---Fetches the name for a given game
----@param args? {game: string?, useDefault: boolean?}
+---@param options? {game: string?, useDefault: boolean?}
 ---@return string?
-function Game.name(args)
-	return Game.raw(args).name
+function Game.name(options)
+	return Game.raw(options).name
 end
 
 ---Fetches the link for a given game
----@param args? {game: string?, useDefault: boolean?}
+---@param options? {game: string?, useDefault: boolean?}
 ---@return string?
-function Game.link(args)
-	return Game.raw(args).link
+function Game.link(options)
+	return Game.raw(options).link
 end
 
 ---Fetches the defaultTeamLogos (light & dark) for a given game
----@param args? {game: string?, useDefault: boolean?}
+---@param options? {game: string?, useDefault: boolean?}
 ---@return table?
-function Game.defaultTeamLogoData(args)
-	return Game.raw(args).defaultTeamLogo
+function Game.defaultTeamLogoData(options)
+	return Game.raw(options).defaultTeamLogo
 end
 
 ---Builds the icon for a given game
----@param args? {game: string?, useDefault: boolean?, size: string?, noLink: boolean?, link: string?}
+---@param options? {game: string?,
+---		useDefault: boolean?,
+---		size: string?,
+---		noLink: boolean?,
+---		link: string?,
+---		noSpan: boolean?,
+---		spanClass: string?,
+---}
```

Comment:
vscode extension doesn't read this correctly iirc (on phone atm)
either in 1 line or make the option params a class in the annotations

Reply:
this

Reply:
https://github.com/Liquipedia/Lua-Modules/blob/07987f01c20a6c853dd385ecbdaed82027bf0f21/standard/flags.lua#L99